### PR TITLE
Updates

### PR DESCRIPTION
**Pull request by AliHajjAli1**
A pr for all my previous tasks

### CODE REVIEW

**Comment by AliHajjAli1** [comment]
Updates for actifit

**Comment by AliHajjAli1** [comment]
I changed the appearance of the day chart button in app/src/main/res/layout/activity_history_single_entry.xml

I created a switch button in the main screen by adding two icons in the app/src/main/res/layout/activity_main.xml, one for each mode. I handled the switch method by clicking them in MainActivity.java

I added a popup for Twitter Actions. This popup appeared when clicking "Actifit on X" button found in the daily reward section. I added app/src/main/res/layout/activity_tweet_actions.xml and handled its method in the MainActivity.java

I added new Activity types in the Post & Earn screen by adding two new strings to the strings.xml and modifying the activity_type array in PostSteemitActivity.java

I added a fitbit sync icon in the activity_main.xml and handled it in the MainActivity.java

**Comment by mcfarhat** [comment]
@AliHajjAli1 please remove config files from the PR, config.properties and local.properties as they can be confusing and disrupt builds elsewhere

File: app/src/main/java/io/actifit/fitnesstracker/actifitfitnesstracker/PostSteemitActivity.java [hunk 1]
```diff
@@ -806,7 +806,7 @@ public void onCheckedChanged(RadioGroup group, int checkedId) {
                 getString(R.string.ChasingPokemons), getString(R.string.Badminton), getString(R.string.PickleBall),
                 getString(R.string.Snowshoeing),getString(R.string.Sailing),getString(R.string.Kayaking), getString(R.string.Kidplay),
                 getString(R.string.HomeImprovement), getString(R.string.YardWork), getString(R.string.StairClimbing),
-                getString(R.string.Yoga), getString(R.string.Stretching), getString(R.string.Plogging), getString(R.string.Crossfit),getString(R.string.FitnessGaming)
+                getString(R.string.Yoga), getString(R.string.Stretching),getString(R.string.Calisthenics),getString(R.string.StreetWorkout), getString(R.string.Plogging), getString(R.string.Crossfit),getString(R.string.Football)
```
Comment (mcfarhat) [inline app/src/main/java/io/actifit/fitnesstracker/actifitfitnesstracker/PostSteemitActivity.java:809]: @AliHajjAli1 it seems you replaced fitnessgaming with football?

File: app/src/main/java/io/actifit/fitnesstracker/actifitfitnesstracker/MainActivity.java [hunk 2]
```diff
@@ -1293,10 +1344,71 @@ public void onClick(DialogInterface dialog, int which) {
...
+                    builder.setPositiveButton("Close", null);
+                    AlertDialog dialog = builder.create();
+                    dialog.show();
+                }
+            });
+
+            freeRewardButton.setOnClickListener(innerView -> showRewardedVideo(innerView, 1,R.id.daily_free_reward,R.id.daily_5k_reward,R.id.daily_7k_reward,R.id.daily_10k_reward));
+            fivekRewardButton.setOnClickListener(innerView -> showRewardedVideo(innerView, 2,R.id.daily_free_reward,R.id.daily_5k_reward,R.id.daily_7k_reward,R.id.daily_10k_reward));
+            sevenkRewardButton.setOnClickListener(innerView -> showRewardedVideo(innerView, 3,R.id.daily_free_reward,R.id.daily_5k_reward,R.id.daily_7k_reward,R.id.daily_10k_reward));
+            tenkRewardButton.setOnClickListener(innerView -> showRewardedVideo(innerView, 4,R.id.daily_free_reward,R.id.daily_5k_reward,R.id.daily_7k_reward,R.id.daily_10k_reward));
```
Comment (mcfarhat) [inline app/src/main/java/io/actifit/fitnesstracker/actifitfitnesstracker/MainActivity.java:1411]: @AliHajjAli1 why did you change the showRewardedVideo function signature and add the 3 params, those params don't change value normally and can be used as constants, as they were. Can you clarify the need for the change?

**Comment by AliHajjAli1** [comment]
> @AliHajjAli1 please remove config files from the PR, config.properties and local.properties as they can be confusing and disrupt builds elsewhere

Okay

File: app/src/main/java/io/actifit/fitnesstracker/actifitfitnesstracker/PostSteemitActivity.java (hunk 1 above)
Comment (AliHajjAli1) [reply app/src/main/java/io/actifit/fitnesstracker/actifitfitnesstracker/PostSteemitActivity.java:809]: Yes, maybe I copy it wrongly, I will fix it.

File: app/src/main/java/io/actifit/fitnesstracker/actifitfitnesstracker/MainActivity.java (hunk 2 above)
Comment (AliHajjAli1) [reply app/src/main/java/io/actifit/fitnesstracker/actifitfitnesstracker/MainActivity.java:1411]: I added those 3 params as the constants, R.id.daily_free_reward,..., were causing an error when using them directly in the method. They were shown as undefined so I put the as params in the method and it works.

File: app/src/main/java/io/actifit/fitnesstracker/actifitfitnesstracker/PostSteemitActivity.java (hunk 1 above)
Comment (mcfarhat) [reply app/src/main/java/io/actifit/fitnesstracker/actifitfitnesstracker/PostSteemitActivity.java]: still not properly typed @AliHajjAli1 , did the project compile for you? 
make sure to follow proper caps/small characters, should be typed as FitnessGaming as this is how its defined in the strings XML file

Comment (AliHajjAli1) [reply app/src/main/java/io/actifit/fitnesstracker/actifitfitnesstracker/PostSteemitActivity.java]: Actually I created a new String so it compiled.
Now I fixed it

File: app/src/main/java/io/actifit/fitnesstracker/actifitfitnesstracker/MainActivity.java [hunk 3]
```diff
@@ -2273,7 +2447,7 @@ private void checkBatteryOptimization(Boolean forceShow) {
 
         PowerManager pm = (PowerManager) getSystemService(Context.POWER_SERVICE);
         FontTextView batteryNotif = findViewById(R.id.battery_notice);
-        if (!pm.isIgnoringBatteryOptimizations("io.actifit.fitnesstracker.actifitfitnesstracker")) {
+        if (!pm.isIgnoringBatteryOptimizations("com.example.trial")) {
```
Comment (mcfarhat) [inline app/src/main/java/io/actifit/fitnesstracker/actifitfitnesstracker/MainActivity.java:2450]: @AliHajjAli1 this is not the correct package name, seems you overwrote the original one "io.actifit.fitnesstracker.actifitfitnesstracke"

File: app/src/main/java/io/actifit/fitnesstracker/actifitfitnesstracker/MainActivity.java [hunk 4]
```diff
@@ -3013,7 +3265,7 @@ public String getFormattedValue(float value, Entry entry, int dataSetIndex, View
 
                 //add limit lines to show marker of min 5K activity
                 //YAxis yAxis = chart.getAxisLeft();
-                dayBarData.setValueFormatter(yFormatter);
+                dayBarData.setValueFormatter((ValueFormatter) yFormatter);
```
Comment (mcfarhat) [inline app/src/main/java/io/actifit/fitnesstracker/actifitfitnesstracker/MainActivity.java:3268]: why aren't you using yformatter defined and initialized above?

File: app/src/main/java/io/actifit/fitnesstracker/actifitfitnesstracker/MainActivity.java [hunk 5]
```diff
@@ -1792,18 +1906,78 @@ public void onClick(DialogInterface dialog, int which) {
...
+                        displayActivityChart(steps, true);
+                        editor.putString("dataTrackingSystem", getString(R.string.device_tracking_ntt));
+                    }
+                    editor.apply();
+                }
+            }
+        });
+
+
+        BtnSwitchSettings.setTooltipText("Switch to Fitbit mode");
```
Comment (mcfarhat) [inline app/src/main/java/io/actifit/fitnesstracker/actifitfitnesstracker/MainActivity.java:1948]: The switch button will always be displayed in the app regardless of the mode. So having a fixed tooltip would be incorrect. Are you taking into consideration the user's current settings for loading the proper mode? the code for this should be in place. I don't think I've seen also any code that stores and saves the selection to the user's settings, or did i miss it? Scenario: user switches to fitbit mode. Closes the app, reopens the app. It should show in fitbit mode.

File: app/src/main/java/io/actifit/fitnesstracker/actifitfitnesstracker/MainActivity.java [hunk 6]
```diff
@@ -4494,7 +4746,7 @@ && checkAppSignature(ctx) == MainActivity.INVALID) {
 
 
                 //make sure package name has not been manipulated
-                if (!ctx.getPackageName().equals("io.actifit.fitnesstracker.actifitfitnesstracker")) {
+                if (!ctx.getPackageName().equals("com.example.trial")) {
```
[the remark this answers is not in the document]
Comment (AliHajjAli1) [reply app/src/main/java/io/actifit/fitnesstracker/actifitfitnesstracker/MainActivity.java:4749]: I will fix them

File: app/src/main/java/io/actifit/fitnesstracker/actifitfitnesstracker/MainActivity.java (hunk 4 above)
Comment (AliHajjAli1) [reply app/src/main/java/io/actifit/fitnesstracker/actifitfitnesstracker/MainActivity.java:3268]: It was showing an error

File: app/src/main/java/io/actifit/fitnesstracker/actifitfitnesstracker/MainActivity.java (hunk 5 above)
Comment (AliHajjAli1) [reply app/src/main/java/io/actifit/fitnesstracker/actifitfitnesstracker/MainActivity.java:1948]: there is 2 tool tip texts, one for each mode
